File: ssort.js

function ssort(arr, n) {
    for (let i=0; i<n-1; i++) {
        let min_idx=i;

        for(let j=i+1; j<n; j++) {
            if(arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        // Swap
        let temp = arr[min_idx];
        arr[min_idx] = arr[i];
        arr[i] = temp;
    }
    return arr;
}


function main() {
    let arr = [5, 3, 7, 2, 9, 1, 0, 6, 3, 5];
    let recv = ssort(arr, arr.length);

    console.log(recv);
}



main();

File: ssort.c
#include <stdio.h>

void selection_sort(int *arr, int n) {

    int min_idx;

    for(int i=0; i<n-1; i++) {
        min_idx = i;
        for(int j=i+1; j<n; j++) {
            if(arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        // Swap the found minimum element with the first element
        int temp = arr[min_idx];
        arr[min_idx] = arr[i];
        arr[i] = temp;
    }
}

int main(int argc, char const *argv[])
{
    int arr[] = {64, 25, 12, 22, 11};
    int n = sizeof(arr) / sizeof(arr[0]);
    selection_sort(arr, n);

    for(int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
    
    return 0;
}
